gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/SubscriberApi.java

```diff
@@ -0,0 +1,663 @@
+   *
+   * @param subscription The subscription whose message is being acknowledged.
+   * @param ackIds The acknowledgment ID for the messages being acknowledged that was returned
+   * by the Pub/Sub system in the Pull response. Must not be empty.
+   */
+  public void acknowledge(String subscription, List<String> ackIds) {
+    AcknowledgeRequest request =
+        AcknowledgeRequest.newBuilder()
+        .setSubscription(subscription)
+        .addAllAckIds(ackIds)
+        .build();
+
+    acknowledge(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Acknowledges the messages associated with the ack tokens in the
+   * AcknowledgeRequest. The Pub/Sub system can remove the relevant messages
+   * from the subscription.
+   *
+   * Acknowledging a message whose ack deadline has expired may succeed,
+   * but such a message may be redelivered later. Acknowledging a message more
+   * than once will not result in an error.
+   *
+   * @param request The request object containing all of the parameters for the API call.
+   */
+  public void acknowledge(AcknowledgeRequest request) {
+    acknowledgeCallable().call(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Acknowledges the messages associated with the ack tokens in the
+   * AcknowledgeRequest. The Pub/Sub system can remove the relevant messages
+   * from the subscription.
+   *
+   * Acknowledging a message whose ack deadline has expired may succeed,
+   * but such a message may be redelivered later. Acknowledging a message more
+   * than once will not result in an error.
+   */
+  public ApiCallable<AcknowledgeRequest, Empty> acknowledgeCallable() {
+    return ACKNOWLEDGE.bind(channel);
+  }
+
+  // ----- pull -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Pulls messages from the server. Returns an empty list if there are no
+   * messages available in the backlog. The server may return UNAVAILABLE if
+   * there are too many concurrent pull requests pending for the given
+   * subscription.
+   *
+   * @param subscription The subscription from which messages should be pulled.
+   * @param returnImmediately If this is specified as true the system will respond immediately even if
+   * it is not able to return a message in the Pull response. Otherwise the
+   * system is allowed to wait until at least one message is available rather
+   * than returning no messages. The client may cancel the request if it does
+   * not wish to wait any longer for the response.
+   * @param maxMessages The maximum number of messages returned for this request. The Pub/Sub
+   * system may return fewer than the number specified.
+   */
+  public PullResponse pull(String subscription, boolean returnImmediately, int maxMessages) {
```

Comment:
How one can "cancel the request if it does not wish to wait any longer for the response"?